subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/ExecutionStateChanges.java

```diff
@@ -28,5 +29,5 @@
      */
     void visitAllChanges(ChangeVisitor visitor);
 
-    InputChangesInternal getInputChanges();
+    InputChangesInternal getInputChanges(ImmutableListMultimap<Object, String> incrementalInputs);
```

Comment:
This is too surprising. We could add some Javadoc here to explain what the `incrementalInputs` parameter does, but I wonder if we could capture what's happening with better names, too.

In general, I think we should reserve the `get` prefix for getters, and use `create` or something else in cases like this.

Reply:
`createInputChanges(incrementalParameterNameByValue)` it is.